arabic/unpacked/arabic.js

```diff
@@ -0,0 +1,448 @@
+          var texCommand = dict[key][0];
+          definitions[texCommand] = key;
+        });
+
+        return definitions;
+      }()
+    });
+
+
+    TEX.Parse.Augment(function () {
+      var parsers = {};
+
+      Object.keys(dict).forEach(function (key) {
+        parsers[key] = dict[key][1]; // Parser function
+      });
+
+      return parsers;
+    }());
+
+
+    TEX.Parse.Augment({
+      flipHorizontal: function (token) {
+        token.arabicFlipH = !token.arabicFlipH;
+          // Invert the value, because flipping twice means, it is not flipped
+        return token;
+      },
+      arabicNumber: function (token) {
+        var numbersMap = MathJax.Hub.config.Arabic.numbersMap;
+        var text = token.data[0].data[0];
+        var mapped = text.replace(englishNumbersRegExp, function (m) {
+          return numbersMap[m];
+        });
+
+        if (mapped !== text) {
+          token.data[0].data[0] = mapped;
+          token.arabicFontLang = 'ar';
+        }
+
+        return this.flipHorizontal(token);
+      },
+      arabicIdentifier: function (token) {
+        var text = token.data[0].data[0];
+
+        if ('chars' === token.data[0].type) {
+          // English Symbols like X and Y
+          var mapped = text.replace(identifiersKeysRegExp, function (m) {
+            return identifiersMap[m.toLowerCase()];
+          });
+
+          if (mapped !== text) {
+            token.data[0].data[0] = mapped;
+            token.arabicFontLang = 'ar';
+          }
+        }
+
+        return this.flipHorizontal(token);
+      },
+      arabicOperator: function (token) {
+        var text = token.data[0].data[0];
+        var mapped = text.replace(operatorsKeysRegExp, function (m) {
+          return operatorsMap[m];
+        });
+
+        if (mapped !== text) {
+          token = this.flipHorizontal(token);
+          token.arabicFontLang = 'ar';
+          token.data[0].data[0] = mapped;
+        }
+
+        return token;
+      },
+      _getArgumentMML: function (name) {
+        //  returns an argument that is a single MathML element
+        //  (in an mrow if necessary)
+        //
+        //  This functions has been copied here from extensions/TeX/HTML.js, to avoid
+        //  adding a dependency.
+        //
+        //  TODO: Consider importing (as a dependency) this from HTML.js instead!
+        var arg = this.ParseArg(name);
+        if (arg.inferred && arg.data.length === 1) {
+          arg = arg.data[0];
+        } else {
+          delete arg.inferred;
+        }
+
+        return arg;
+      },
+      Push: function () {
+        var beforeLang = this.stack.env.lang;
+        var retVal = texParsePush.apply(this, arguments);
+
+        // Huge giant hack to propagate `lang` from Arrays (and other environments?)
+        // to their children fractions and others.
+        if (beforeLang) {
+          if (!this.stack.env.lang) {
+            this.stack.env.lang = beforeLang;
+          }
+        }
```

Comment:
Thanks for all your work on this extension, and for being willing to adjust things based on my comments.  It has been a pleasure to work with you on this project.